arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentCollectionViewController.swift

```diff
@@ -141,24 +119,27 @@ class ContentCollectionViewController: UIViewController, UICollectionViewDataSou
     
     //MARK: - UICollectionViewDelegate
     
-    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
         //hide keyboard if visible
         self.view.endEditing(true)
         
         let category = categories[indexPath.item]
-        let controller = self.storyboard!.instantiateViewController(withIdentifier: "ContentTableViewController") as! ContentTableViewController
+        let controller = storyboard!.instantiateViewController(withIdentifier: "ContentTableViewController") as! ContentTableViewController
         controller.allSamples = category.samples
         controller.title = category.name
-        self.navigationController?.show(controller, sender: self)
+        navigationController?.show(controller, sender: self)
     }
     
     //MARK: - UICollectionViewDelegateFlowLayout
     
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        
-        let size = self.itemSizeForCollectionViewSize(self.collectionView.bounds.size)
-        
-        return size
+        var collectionViewSize = collectionView.bounds.size
+        if #available(iOS 11.0, *) {
+            //account for the safe area when determining the item size
+            collectionViewSize.height -= collectionView.safeAreaInsets.bottom + collectionView.safeAreaInsets.top
+            collectionViewSize.width -= collectionView.safeAreaInsets.left + collectionView.safeAreaInsets.right
+        }
+        return itemSizeForCollectionViewSize(collectionViewSize)
```

Comment:
Since `itemSizeForCollectionViewSize(_:)` is only called from here, and because the method is doing what this delegate callback is supposed to be doing, consider just inlining it here. I'm not sure there is any benefit to having it as a separate method.